test/browser/fragments.test.js

```diff
@@ -237,7 +237,9 @@ describe('Fragment', () => {
 		expectDomLogToBe([
 			'<div>.appendChild(#text)',
 			'<div>122.insertBefore(<div>1, <span>1)',
-			'<span>1.remove()'
+			'<span>1.remove()',
+			'<div>122.appendChild(<span>2)',
```

Comment:
Yea, this was a performance optimization but it breaks unexpectedly. I can look into building some offset based calculator but _generally_ this optimization is impossible to get right because of several reasons

- we can have nullish children
- we can have empty fragments as children

both these conditions increment the i counter meaning we will start looking for a dom-sibling in an erroneous position, hence the optimization fails in the issue-case and multiple others 😅

Reply:
Also do read that the DOM is still correct we just can't prevent these dupe insertions when moving

Reply:
Agree, that this seems to point at a shortcoming of the current reconciler algo.